src/Providers/AuthUserProvider.tsx

```diff
@@ -70,15 +85,46 @@ export default function AuthUserProvider({
 
   const { mutateAsync: signIn, isPending: isAuthenticating } = useMutation({
     mutationFn: mutate(routes.login),
-    onSuccess: (data: JwtTokenObtainPair) => {
+    onSuccess: async (data: LoginResponse) => {
+      // Handle MFA cases
+      if (isMFAResponse(data)) {
+        localStorage.setItem("mfa_temp_token", data.temp_token);
+        navigate("/2fa");
+        return;
+      }
+
+      // Handle normal login with JWT tokens
+      if (isJwtTokenResponse(data)) {
+        setAccessToken(data.access);
+        localStorage.setItem(LocalStorageKeys.accessToken, data.access);
+        localStorage.setItem(LocalStorageKeys.refreshToken, data.refresh);
+
+        await queryClient.invalidateQueries({ queryKey: ["currentUser"] });
+
+        if (location.pathname === "/" || location.pathname === "/login") {
+          navigate(getRedirectOr("/"));
+        }
+      }
+    },
+  });
+
+  const { mutateAsync: verifyMFA, isPending: isVerifyingMFA } = useMutation({
+    mutationFn: mutate(authApi.mfa.login),
+    onSuccess: async (data: JwtTokenObtainPair) => {
+      // Clear MFA related data
+      localStorage.removeItem("mfa_temp_token");
+
+      // Set new tokens
       setAccessToken(data.access);
       localStorage.setItem(LocalStorageKeys.accessToken, data.access);
       localStorage.setItem(LocalStorageKeys.refreshToken, data.refresh);
-      queryClient.invalidateQueries({ queryKey: ["currentUser"] });
 
-      if (location.pathname === "/" || location.pathname === "/login") {
-        navigate(getRedirectOr("/"));
-      }
+      // Invalidate and wait for the currentUser query to complete
+      await queryClient.invalidateQueries({ queryKey: ["currentUser"] });
+
+      // Get redirect URL from query params or localStorage
+      const redirectURL = getRedirectURL() || "/";
+      navigate(redirectURL);
```

Comment:
```suggestion
      navigate(getRedirectOr("/"));
```

This method already taking care https://github.com/abhimanyurajeesh/care_fe/blob/b2c3d26a07c042f952d61878182c695647bdd26c/src/Providers/AuthUserProvider.tsx#L223

Reply:
done